### PR TITLE
driver: i2s: mcux_i2s_sai fixing build error

### PR DESCRIPTION
**Pull request by hakehuang**
there some build error due to add rt11xx support without testing.

still there are some other PRs to full enable i2s on rt platform

this fixing: #41153

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
i guess this means we need to ensure a build target with right configuration to ensure this builds in the future?

**Comment by lgl88911** [comment]
@hakehuang  It seems that the following content needs to be removed to build the pass. Or am I missing something else that caused the pinmux build error here?
```@@ -1117,20 +1116,11 @@ static const struct i2s_driver_api i2s_mcux_driver_api = {                .pll_den =                                              \                        DT_PHA_BY_NAME(DT_DRV_INST(i2s_id),             \                                pll_clocks, den, value),                \-               .mclk_pin_mask =                                        \-                       DT_PHA_BY_IDX(DT_DRV_INST(i2s_id),              \-                               pinmuxes, 0, function),                 \-               .mclk_pin_offset =                                      \-                       DT_PHA_BY_IDX(DT_DRV_INST(i2s_id),              \-                               pinmuxs, 0, pin),                       \                .clk_sub_sys =  (clock_control_subsys_t)                \                        DT_CLOCKS_CELL_BY_IDX(DT_DRV_INST(i2s_id),      \                                0, name),                               \                .ccm_dev = DEVICE_DT_GET(                               \                                DT_CLOCKS_CTLR(DT_DRV_INST(i2s_id))),   \-               .pinmux_dev = DEVICE_DT_GET(                            \-                               DT_PHANDLE_BY_IDX(DT_DRV_INST(i2s_id),  \-                               pinmuxes, 0)),                          \                .irq_connect = i2s_irq_connect_##i2s_id,                \```

```zephyr/drivers/i2s/i2s_mcux_sai.c: At top level:build/zephyr/include/generated/devicetree_unfixed.h:3977:34: error: 'DT_N_S_soc_S_sai_40384000_P_pinmuxs_IDX_0_VAL_pin' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_sai_40384000_P_pinmuxes_IDX_0_VAL_pin'? 3977 | #define DT_N_INST_0_nxp_mcux_i2s DT_N_S_soc_S_sai_40384000```

**Comment by hakehuang** [comment]
@lgl88911 , this is required to set mclk, you can remove it at your side first, I have some following PR to add pinctrl support, so that the mclck can be set.

**Comment by lgl88911** [comment]
@hakehuang Got it, Thanks！Looks like it will take a while

**Review by mmahadevan108 — APPROVED** [review]

File: drivers/i2s/i2s_mcux_sai.c [hunk 1]
```diff
@@ -333,21 +332,19 @@ static void enable_mclk_direction(const struct device *dev, bool dir)
 	uint32_t offset = dev_cfg->mclk_pin_offset;
 	uint32_t mask = dev_cfg->mclk_pin_mask;
 	uint32_t value = 0;
+	uint32_t *gpr = (uint32_t *)DT_REG_ADDR(DT_NODELABEL(iomuxcgpr)) + offset;
```
Comment (gmarull) [inline drivers/i2s/i2s_mcux_sai.c:335]: where is the DT node defined now?

File: drivers/i2s/i2s_mcux_sai.c [hunk 2]
```diff
@@ -333,21 +332,19 @@ static void enable_mclk_direction(const struct device *dev, bool dir)
...
 
 	if (iomuxgpr_dev == NULL) {
 		LOG_ERR("fail to find pinmux controller\n");
 		return;
 	}
```
Comment (gmarull) [inline drivers/i2s/i2s_mcux_sai.c:340]: why is this required now?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/i2s/i2s_mcux_sai.c (hunk 2 above)
Comment (hakehuang) [reply drivers/i2s/i2s_mcux_sai.c:340]: @gmarull if not defined it is NULL, and access this will have segmetation fault

File: drivers/i2s/i2s_mcux_sai.c (hunk 1 above)
Comment (hakehuang) [reply drivers/i2s/i2s_mcux_sai.c:335]: @gmarull this is defined in the [here](iomuxcgpr: iomuxcgpr@400ac000 {)

File: drivers/i2s/i2s_mcux_sai.c (hunk 2 above)
Comment (gmarull) [reply drivers/i2s/i2s_mcux_sai.c:340]: this file is no longer using a pinmux device, so please delete all related code.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/i2s/i2s_mcux_sai.c (hunk 2 above)
Comment (hakehuang) [reply drivers/i2s/i2s_mcux_sai.c]: updated

File: drivers/i2s/i2s_mcux_sai.c [hunk 3]
```diff
@@ -333,21 +332,20 @@ static void enable_mclk_direction(const struct device *dev, bool dir)
...
 	uint32_t value = 0;
+	uint32_t *gpr_base = (uint32_t *)DT_REG_ADDR(DT_NODELABEL(iomuxcgpr));
+	uint32_t *gpr =  gpr_base + offset;
 
-	if (iomuxgpr_dev == NULL) {
-		LOG_ERR("fail to find pinmux controller\n");
+	if (gpr_base == NULL) {
+		LOG_ERR("fail to find iomuxcgpr base\n");
 		return;
 	}
```
Comment (gmarull) [inline drivers/i2s/i2s_mcux_sai.c:341]: what is the point of this check? when is this true?

**Review by gmarull — DISMISSED** [review]
all pinmux code needs to be deleted now, including related config fields, includes, etc.

File: drivers/i2s/i2s_mcux_sai.c (hunk 3 above)
Comment (hakehuang) [reply drivers/i2s/i2s_mcux_sai.c:341]: in case user does not define the iomuxgor then the (uint32_t *)DT_REG_ADDR(DT_NODELABEL(iomuxcgpr)) wil be NULL right?

Comment (gmarull) [reply drivers/i2s/i2s_mcux_sai.c:341]: no: build will fail.

Comment (hakehuang) [reply drivers/i2s/i2s_mcux_sai.c:341]: ok, let me remove this line.

File: drivers/i2s/i2s_mcux_sai.c (hunk 2 above)
Comment (gmarull) [reply drivers/i2s/i2s_mcux_sai.c]: please, delete all pinmux related code: include, unused config fields

**Comment by hakehuang** [comment]
@gmarull can help review again?

**Review by carlescufi — APPROVED** [review]
